File: Data Structures & Algorithms/validate-parentheses/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const stack = new Array();
        for (let i = 0; i < s.length; i++){
            if (s[i] === "("){
                stack.push(s[i]);
            }
            else if (s[i] === "["){
                stack.push(s[i]);
            }
            else if (s[i] === "{"){
                stack.push(s[i]);
            }
            else if (s[i] === ")"){
                let top = stack.pop();
                if (top !== "(" ){
                    return false;
                }
            }
            else if (s[i] === "]"){
                let top = stack.pop();
                if (top !== "[" ){
                    return false;
                }
            }
            else if (s[i] === "}"){
                let top = stack.pop();
                if (top !== "{" ){
                    return false;
                }
            }
            

        }
        if (stack.length === 0){
            return true
        }
        return false;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        res = 0
        for token in tokens:
            if token.isdigit():
                stack.append(int(token))
            else:
                b = stack.pop()
                a = stack.pop()
                stack.append(eval(f"{a} {token} {b}"))
        return stack.pop()
